tests/test_routines.py

```diff
@@ -38,7 +38,7 @@ def test_load_arduino():
 def test_initialize_arduino():
     da = Dummy_arduino()
     da.name = "good_arduino"
-
+    da.firmware = "pytentiostat_firmata.ino"
```

Comment:
this doesn't seem to be tested.  We would need some assert statements that test different scenarios.

Reply:
I pushed a new version with an assert statement to test. I think that the negative check has been already covered in line 42 of test_routines.py,  I'm wrong?

Reply:
you are testing desired behavior of the app.  So answer these questions:
What do you want the program to do if da.firmware is the wrong value?  
or None?
or even the right value?  Right now we are not really testing anything because we set an attribute to something then ask if it is that thing.

Reply:
Now I think I got it. Create two different instances, one that has the correct firmware and one that doesn't. The good arduino should return his name and the bad one should raise an error.